cli/internal/context/context.go

```diff
@@ -315,17 +312,6 @@ func (c *Context) populateWorkspaceGraphForPackageJSON(pkg *fs.PackageJSON, root
 		}
 	}
 
-	externalDeps, err := lockfile.TransitiveClosure(
-		pkg.Dir.ToUnixPath(),
-		pkg.UnresolvedExternalDeps,
-		c.Lockfile,
-	)
-	if err != nil {
-		warnings.append(err)
-		// reset external deps to original state
-		externalDeps = mapset.NewSet()
```

Comment:
This was here instead of proper control flow, if the closure calculation errored we would override the return value with an empty set. This would then get translated to `pkg.TransitiveDeps = make([]lockfile.Package, 0)` and nothing would get added.